mempool/src/shared_mempool.rs

```diff
@@ -96,7 +97,7 @@ where
     config: MempoolConfig,
     network_senders: HashMap<PeerId, MempoolNetworkSender>,
     storage_read_client: Arc<dyn StorageRead>,
-    validator: Arc<V>,
+    validator: Arc<RwLock<V>>,
```

Comment:
just curious, does it have any perf impact?
I took a look in a code. There's no need for RwLock
Validator is currently part of SharedMempool struct, but it doesn't have to be
Alternatively, it can just be local variable in inbound_network_task. 
Then you can just send message to inbound_network_task to recreate or restart vm_validator(e.g. update local variable).
And then inbound_network_task will send new clone to process_incoming_transactions